# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: accessible/base/nsAccUtils.cpp
# Commit: c0a7fdc4bfca
# Full Hash: c0a7fdc4bfca0596ffecd347f0e248440e371989
# Author: Eitan Isaacson <eitan@monotonous.org>
# Date: 2022-01-05 09:33:53
# Regressor Bug: 1737944
# File Overlap Count: 2
# Description:
#   Bug 1737944 - P5: Unify GetLevelInternal and GetDefaultLevel as Accessible::GetLevel. r=Jamie
#   
#   The nsAccUtils method was a "fast" one for calculating set sizes and
#   conceptual parents. Unified it with
#   LocalAccessible::GetLevelInternal in Accessible::GetLevel with an
# ==============================================================================

diff -r f6ce0e05c92e -r c0a7fdc4bfca accessible/base/nsAccUtils.cpp
--- a/accessible/base/nsAccUtils.cpp	Tue Jan 04 21:01:35 2022 +0000
+++ b/accessible/base/nsAccUtils.cpp	Tue Jan 04 21:01:36 2022 +0000
@@ -43,31 +43,6 @@
   }
 }
 
-int32_t nsAccUtils::GetDefaultLevel(const LocalAccessible* aAccessible) {
-  roles::Role role = aAccessible->Role();
-
-  if (role == roles::OUTLINEITEM) return 1;
-
-  if (role == roles::ROW) {
-    LocalAccessible* parent = aAccessible->LocalParent();
-    // It is a row inside flatten treegrid. Group level is always 1 until it
-    // is overriden by aria-level attribute.
-    if (parent && parent->Role() == roles::TREE_TABLE) return 1;
-  }
-
-  return 0;
-}
-
-int32_t nsAccUtils::GetARIAOrDefaultLevel(const LocalAccessible* aAccessible) {
-  int32_t level = 0;
-  nsCoreUtils::GetUIntAttr(aAccessible->GetContent(), nsGkAtoms::aria_level,
-                           &level);
-
-  if (level != 0) return level;
-
-  return GetDefaultLevel(aAccessible);
-}
-
 int32_t nsAccUtils::GetLevelForXULContainerItem(nsIContent* aContent) {
   nsCOMPtr<nsIDOMXULContainerItemElement> item =
       aContent->AsElement()->AsXULContainerItem();